imagetest/test_suites/security/image_security_test.go

```diff
@@ -0,0 +1,341 @@
+			t.Fatal(err)
+		}
+	case strings.Contains(image, "debian") || strings.Contains(image, "ubuntu"):
+		if err := verifyPackageInstalled(); err != nil {
+			t.Fatal(err)
+		}
+		// Check that the security packages are marked for automatic update
+		// https://wiki.debian.org/UnattendedUpgrades
+		// https://help.ubuntu.com/community/AutomaticSecurityUpdates
+		if err := verifySecurityUpgradeEnabled(image); err != nil {
+			t.Fatal(err)
+		}
+		if err := verifyAutomaticUpdate(image); err != nil {
+			t.Fatal(err)
+		}
+	case strings.Contains(image, "sles"):
+		t.Skip("skipping test on openSUSE.")
+	}
+}
+
+// TestPasswordSecurity Ensure that the system enforces strong passwords and correct lockouts.
+func TestPasswordSecurity(t *testing.T) {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+	if strings.Contains(image, "rhel") {
+		if err := verifyCracklibInstalled(); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	if err := verifySSHConfig(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Root password/login is disabled.
+	if err := verifyPassword(); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func verifyPassword() error {
+	fileBytes, err := ioutil.ReadFile("/etc/passwd")
+	if err != nil {
+		return err
+	}
+	passwards := strings.Split(string(fileBytes), "\n")
+	for _, pwd := range passwards {
+		// ignore empty pwd
+		if len(pwd) == 0 {
+			continue
+		}
+		passwdItems := strings.Split(pwd, ":")
+		loginname := passwdItems[0]
+		passwd := passwdItems[1]
+		uid := passwdItems[2]
+		shell := passwdItems[6]
+		// A passwd entry of length 1 is a disabled password.
+		if len(passwd) != 1 {
+			return fmt.Errorf("login for %s does not look disabled", loginname)
+		}
+
+		uidValue, err := strconv.Atoi(uid)
+		if err != nil {
+			return err
+		}
+		// Don't check user account and root account
+		if uidValue >= minUID || uidValue == 0 {
+			continue
+		}
+		if targetShell, found := specialAccountShells[loginname]; found {
+			if targetShell != shell {
+				return fmt.Errorf("account %s has wrong login shell %s", loginname, shell)
+			}
+		}
+		if !strings.Contains(shell, "false") && !strings.Contains(shell, "nologin") {
+			return fmt.Errorf("account %s has the login shell %s", loginname, shell)
+		}
+	}
+	return nil
+}
+
+func verifySSHConfig() error {
+	fileBytes, err := ioutil.ReadFile("/etc/ssh/sshd_config")
+	if err != nil {
+		return err
+	}
+	sshdConfig := string(fileBytes)
+	if !strings.Contains(sshdConfig, "PasswordAuthentication no") {
+		return fmt.Errorf("\"PasswordAuthentication\" was not set to \"no\"")
+	}
+
+	noRootSSH := strings.Contains(sshdConfig, "PermitRootLogin no")
+	noRootPasswordSSH := strings.Contains(sshdConfig, "PermitRootLogin without-password")
+	if !noRootSSH && !noRootPasswordSSH {
+		return fmt.Errorf("\"PermitRootLogin\" was not set to \"no\" or \"without-password\"")
+	}
+	return nil
+}
+
+func verifyCracklibInstalled() error {
```

Comment:
I'm not sure why we care about this actually. We don't use passwords for anything and there's no reason to check for this. I say remove it. @hopkiw do you agree?

Reply:
we can't remove the cracklib package (pam depends on it), but i am unsure why we test for it specifically. if you don't know either, i'm fine removing it.

Reply:
Oh yeah- not suggesting we remove it. Just suggesting we remove this test case (I have no idea why we have it).